### PR TITLE
feature: update internet_max_bandwidth_out with no forceNew operation

### PR DESCRIPTION
**Pull request by piggona**
go test -v ./tencentcloud/ -test.run TestAccTencentCloudInstanceWithNetwork
=== RUN   TestAccTencentCloudInstanceWithNetwork
=== PAUSE TestAccTencentCloudInstanceWithNetwork
=== CONT  TestAccTencentCloudInstanceWithNetwork
--- PASS: TestAccTencentCloudInstanceWithNetwork (88.49s)
PASS
ok      github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud     88.500s

### CODE REVIEW

File: tencentcloud/resource_tc_instance.go [hunk 1]
```diff
@@ -943,6 +943,33 @@ func resourceTencentCloudInstanceUpdate(d *schema.ResourceData, meta interface{}
...
+		if chargeType := d.Get("internet_charge_type").(string); chargeType != "TRAFFIC_POSTPAID_BY_HOUR" && chargeType != "BANDWIDTH_POSTPAID_BY_HOUR" && chargeType != "BANDWIDTH_PACKAGE" {
+			return fmt.Errorf("bad internet_charge_type.type should be in: TRAFFIC_POSTPAID_BY_HOUR BANDWIDTH_POSTPAID_BY_HOUR BANDWIDTH_PACKAGE")
+		}
+
+		err := cvmService.ModifyInternetMaxBandwidthOut(ctx, instanceId, d.Get("internet_charge_type").(string), d.Get("internet_max_bandwidth_out").(int64))
+		if err != nil {
+			return err
+		}
+		d.SetPartial("internet_max_bandwidth_out")
+		time.Sleep(10 * time.Second)
```
Comment (crab21) [inline tencentcloud/resource_tc_instance.go:956]: 为何sleep这么久

File: tencentcloud/service_tencentcloud_cvm.go [hunk 2]
```diff
@@ -179,6 +179,28 @@ func (me *CvmService) ModifyPassword(ctx context.Context, instanceId, password s
...
+	}
+
+	ratelimit.Check(request.GetAction())
+	response, err := me.client.UseCvmClient().ResetInstancesInternetMaxBandwidth(request)
+	if err != nil {
+		log.Printf("[CRITAL]%s api[%s] fail, request body [%s], reason[%s]\n",
+			logId, request.GetAction(), request.ToJsonString(), err.Error())
+		return err
+	}
+	log.Printf("[DEBUG]%s api[%s] success, request body [%s], response body [%s]\n",
```
Comment (crab21) [inline tencentcloud/service_tencentcloud_cvm.go:198]: 正常逻辑不用这个日志

Comment (piggona) [reply tencentcloud/service_tencentcloud_cvm.go]: fixed

File: tencentcloud/resource_tc_instance.go (hunk 1 above)
Comment (piggona) [reply tencentcloud/resource_tc_instance.go]: fixed

File: tencentcloud/resource_tc_instance.go [hunk 3]
```diff
@@ -943,6 +942,32 @@ func resourceTencentCloudInstanceUpdate(d *schema.ResourceData, meta interface{}
 		d.SetPartial("tags")
 	}
 
+	if d.HasChange("internet_max_bandwidth_out") {
+		if chargeType := d.Get("internet_charge_type").(string); chargeType != "TRAFFIC_POSTPAID_BY_HOUR" && chargeType != "BANDWIDTH_POSTPAID_BY_HOUR" && chargeType != "BANDWIDTH_PACKAGE" {
+			return fmt.Errorf("bad internet_charge_type.type should be in: TRAFFIC_POSTPAID_BY_HOUR BANDWIDTH_POSTPAID_BY_HOUR BANDWIDTH_PACKAGE")
```
Comment (oliverpei) [inline tencentcloud/resource_tc_instance.go:947]: 提示改成：只支持TRAFFIC_POSTPAID_BY_HOUR， BANDWIDTH_POSTPAID_BY_HOUR， BANDWIDTH_PACKAGE计费方式的带宽修改

File: tencentcloud/resource_tc_instance.go [hunk 4]
```diff
@@ -943,6 +942,32 @@ func resourceTencentCloudInstanceUpdate(d *schema.ResourceData, meta interface{}
...
+		d.SetPartial("internet_max_bandwidth_out")
+		err = resource.Retry(2*readRetryTimeout, func() *resource.RetryError {
+			instance, errRet := cvmService.DescribeInstanceById(ctx, instanceId)
+			if errRet != nil {
+				return retryError(errRet, InternalError)
+			}
+			if instance != nil && *instance.LatestOperationState == CVM_LATEST_OPERATION_STATE_OPERATING {
+				return resource.RetryableError(fmt.Errorf("cvm instance latest operetion status is %s, retry...", *instance.LatestOperationState))
+			}
+			return nil
```
Comment (Himer) [inline tencentcloud/resource_tc_instance.go:963]: state=="success" return nil 并d.SetPartial("internet_max_bandwidth_out")
state=="fail"    return error吧 ?

File: tencentcloud/resource_tc_instance.go [hunk 5]
```diff
@@ -200,7 +200,6 @@ func resourceTencentCloudInstance() *schema.Resource {
 				Type:        schema.TypeInt,
 				Optional:    true,
 				Computed:    true,
-				ForceNew:    true,
```
Comment (Himer) [inline tencentcloud/resource_tc_instance.go:203]: changelog中需要标识字段从ForceNew变成可修改 (在什么情况下)

File: tencentcloud/resource_tc_instance.go (hunk 3 above)
Comment (gailwang) [inline tencentcloud/resource_tc_instance.go:947]: 报错不够清晰，什么场景下必须遵守这个范围限制

File: tencentcloud/resource_tc_instance.go [hunk 6]
```diff
@@ -943,6 +942,32 @@ func resourceTencentCloudInstanceUpdate(d *schema.ResourceData, meta interface{}
 		d.SetPartial("tags")
 	}
 
+	if d.HasChange("internet_max_bandwidth_out") {
+		if chargeType := d.Get("internet_charge_type").(string); chargeType != "TRAFFIC_POSTPAID_BY_HOUR" && chargeType != "BANDWIDTH_POSTPAID_BY_HOUR" && chargeType != "BANDWIDTH_PACKAGE" {
+			return fmt.Errorf("bad internet_charge_type.type should be in: TRAFFIC_POSTPAID_BY_HOUR BANDWIDTH_POSTPAID_BY_HOUR BANDWIDTH_PACKAGE")
+		}
+
+		err := cvmService.ModifyInternetMaxBandwidthOut(ctx, instanceId, d.Get("internet_charge_type").(string), d.Get("internet_max_bandwidth_out").(int64))
```
Comment (gailwang) [inline tencentcloud/resource_tc_instance.go:950]: 这个用到多次，简化成一个变量

File: tencentcloud/resource_tc_instance.go (hunk 5 above)
Comment (oliverpei) [inline tencentcloud/resource_tc_instance.go:203]: 执行下make doc, 文档有修改

File: tencentcloud/resource_tc_instance.go (hunk 3 above)
Comment (piggona) [reply tencentcloud/resource_tc_instance.go:947]: fixed

Comment (piggona) [reply tencentcloud/resource_tc_instance.go:947]: fixed

File: tencentcloud/resource_tc_instance.go (hunk 5 above)
Comment (piggona) [reply tencentcloud/resource_tc_instance.go:203]: fixed

File: tencentcloud/resource_tc_instance.go (hunk 6 above)
Comment (piggona) [reply tencentcloud/resource_tc_instance.go]: handled

File: tencentcloud/resource_tc_instance.go (hunk 5 above)
Comment (piggona) [reply tencentcloud/resource_tc_instance.go:203]: fixed

File: tencentcloud/resource_tc_instance.go (hunk 4 above)
Comment (piggona) [reply tencentcloud/resource_tc_instance.go:965]: fixed

**Comment by piggona** [comment]
terraform apply
tencentcloud_security_group.default: Refreshing state... [id=sg-20zxkald]
data.tencentcloud_availability_zones.default: Refreshing state...
tencentcloud_eip.haohao: Refreshing state... [id=eip-ax4g081u]
data.tencentcloud_images.default: Refreshing state...
data.tencentcloud_instance_types.default: Refreshing state...
tencentcloud_security_group_rule.ssh: Refreshing state... [id=eyJzZ19pZCI6InNnLTIwenhrYWxkIiwicG9saWN5X3R5cGUiOiJpbmdyZXNzIiwiY2lkcl9pcCI6IjAuMC4wLjAvMCIsInByb3RvY29sIjoidGNwIiwicG9ydF9yYW5nZSI6IjIyIiwiYWN0aW9uIjoiYWNjZXB0Iiwic291cmNlX3NnX2lkIjoiIiwiZGVzY3JpcHRpb24iOiIifQ==]
tencentcloud_security_group_rule.web: Refreshing state... [id=eyJzZ19pZCI6InNnLTIwenhrYWxkIiwicG9saWN5X3R5cGUiOiJpbmdyZXNzIiwiY2lkcl9pcCI6IjAuMC4wLjAvMCIsInByb3RvY29sIjoidGNwIiwicG9ydF9yYW5nZSI6IjgwLDgwODAiLCJhY3Rpb24iOiJhY2NlcHQiLCJzb3VyY2Vfc2dfaWQiOiIiLCJkZXNjcmlwdGlvbiI6IiJ9]
tencentcloud_instance.web[0]: Refreshing state... [id=ins-ifh44bjc]
tencentcloud_eip_association.haohao: Refreshing state... [id=eip-ax4g081u::ins-ifh44bjc]

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # tencentcloud_instance.web[0] will be updated in-place
  ~ resource "tencentcloud_instance" "web" {
        allocate_public_ip         = true
        availability_zone          = "ap-guangzhou-3"
        create_time                = "2020-08-27T07:59:46Z"
        disable_monitor_service    = false
        disable_security_service   = false
        force_delete               = false
        id                         = "ins-ifh44bjc"
        image_id                   = "img-25szkc8t"
        instance_charge_type       = "POSTPAID_BY_HOUR"
        instance_name              = "yuanhaojin test server"
        instance_status            = "RUNNING"
        instance_type              = "S1.SMALL1"
        internet_charge_type       = "TRAFFIC_POSTPAID_BY_HOUR"
      ~ internet_max_bandwidth_out = 1 -> 5
        private_ip                 = "10.104.18.211"
        project_id                 = 0
        public_ip                  = "119.29.77.196"
        running_flag               = true
        security_groups            = [
            "sg-20zxkald",
        ]
        system_disk_id             = "disk-3pp2ysbm"
        system_disk_size           = 50
        system_disk_type           = "CLOUD_PREMIUM"
        tags                       = {
            "purpose" = "for test"
            "who"     = "yuanhaojin"
        }
    }

Plan: 0 to add, 1 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

tencentcloud_instance.web[0]: Modifying... [id=ins-ifh44bjc]
tencentcloud_instance.web[0]: Modifications complete after 7s [id=ins-ifh44bjc]

Apply complete! Resources: 0 added, 1 changed, 0 destroyed.

**Comment by piggona** [comment]
terraform apply
```
tencentcloud_security_group.default: Refreshing state... [id=sg-20zxkald]
data.tencentcloud_availability_zones.default: Refreshing state...
tencentcloud_eip.haohao: Refreshing state... [id=eip-ax4g081u]
data.tencentcloud_images.default: Refreshing state...
data.tencentcloud_instance_types.default: Refreshing state...
tencentcloud_security_group_rule.ssh: Refreshing state... [id=eyJzZ19pZCI6InNnLTIwenhrYWxkIiwicG9saWN5X3R5cGUiOiJpbmdyZXNzIiwiY2lkcl9pcCI6IjAuMC4wLjAvMCIsInByb3RvY29sIjoidGNwIiwicG9ydF9yYW5nZSI6IjIyIiwiYWN0aW9uIjoiYWNjZXB0Iiwic291cmNlX3NnX2lkIjoiIiwiZGVzY3JpcHRpb24iOiIifQ==]
tencentcloud_security_group_rule.web: Refreshing state... [id=eyJzZ19pZCI6InNnLTIwenhrYWxkIiwicG9saWN5X3R5cGUiOiJpbmdyZXNzIiwiY2lkcl9pcCI6IjAuMC4wLjAvMCIsInByb3RvY29sIjoidGNwIiwicG9ydF9yYW5nZSI6IjgwLDgwODAiLCJhY3Rpb24iOiJhY2NlcHQiLCJzb3VyY2Vfc2dfaWQiOiIiLCJkZXNjcmlwdGlvbiI6IiJ9]
tencentcloud_instance.web[0]: Refreshing state... [id=ins-ifh44bjc]
tencentcloud_eip_association.haohao: Refreshing state... [id=eip-ax4g081u::ins-ifh44bjc]

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # tencentcloud_instance.web[0] will be updated in-place
  ~ resource "tencentcloud_instance" "web" {
        allocate_public_ip         = true
        availability_zone          = "ap-guangzhou-3"
        create_time                = "2020-08-27T07:59:46Z"
        disable_monitor_service    = false
        disable_security_service   = false
        force_delete               = false
        id                         = "ins-ifh44bjc"
        image_id                   = "img-25szkc8t"
        instance_charge_type       = "POSTPAID_BY_HOUR"
        instance_name              = "yuanhaojin test server"
        instance_status            = "RUNNING"
        instance_type              = "S1.SMALL1"
        internet_charge_type       = "TRAFFIC_POSTPAID_BY_HOUR"
      ~ internet_max_bandwidth_out = 1 -> 5
        private_ip                 = "10.104.18.211"
        project_id                 = 0
        public_ip                  = "119.29.77.196"
        running_flag               = true
        security_groups            = [
            "sg-20zxkald",
        ]
        system_disk_id             = "disk-3pp2ysbm"
        system_disk_size           = 50
        system_disk_type           = "CLOUD_PREMIUM"
        tags                       = {
            "purpose" = "for test"
            "who"     = "yuanhaojin"
        }
    }

Plan: 0 to add, 1 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

tencentcloud_instance.web[0]: Modifying... [id=ins-ifh44bjc]
tencentcloud_instance.web[0]: Modifications complete after 7s [id=ins-ifh44bjc]

Apply complete! Resources: 0 added, 1 changed, 0 destroyed.
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -5,6 +5,10 @@ BUG FIXES:
 * Resource: `tencentcloud_vpn_gateway` fix force new issue when apply repeatedly.
 * Resource: `tencentcloud_vpn_connection` fix force new issue when apply repeatedly.
 
+ENHANCEMENTS:
+
+* Resource: `tencentcloud_instance` support for adjusting `internet_max_bandwidth_out` without forceNew when attribute `internet_charge_type` within `TRAFFIC_POSTPAID_BY_HOUR` `BANDWIDTH_POSTPAID_BY_HOUR` `BANDWIDTH_PACKAGE`.
```
Comment (gailwang) [inline CHANGELOG.md:10]: 逗号呢

File: CHANGELOG.md [hunk 8]
```diff
@@ -5,6 +5,10 @@ BUG FIXES:
 * Resource: `tencentcloud_vpn_gateway` fix force new issue when apply repeatedly.
 * Resource: `tencentcloud_vpn_connection` fix force new issue when apply repeatedly.
```
Comment (gailwang) [inline CHANGELOG.md:7]: fix了哪个bug也写上

**Review by gailwang — APPROVED** [review]

**Review by oliverpei — APPROVED** [review]

**Review by crab21 — APPROVED** [review]

**Pull request merged**